crates/extensions/aws_mapper_ext/src/converter.rs

```diff
@@ -41,45 +45,104 @@ impl AwsConverter {
+        input: &MqttMessage,
+    ) -> Result<Vec<MqttMessage>, ConversionError> {
+        let default_timestamp = self.add_timestamp.then(|| self.clock.now());
+
+        // serialize with ThinEdgeJson for measurements, for alarms and events just add the timestamp
+        if input.topic.name.starts_with("tedge/health") {
+            let mut payload_json: Map<String, Value> =
+                serde_json::from_slice(input.payload.as_bytes())?;
+            if !payload_json.contains_key("time") {
                 if let Some(timestamp) = default_timestamp {
                     let timestamp = timestamp
                         .format(&time::format_description::well_known::Rfc3339)?
                         .as_str()
                         .into();
                     payload_json.entry("time").or_insert(timestamp);
                 }
-
-                serde_json::to_string(&payload_json)?
-            } else {
-                return Ok(vec![]);
-            };
-
+            }
+            let payload = serde_json::to_string(&payload_json)?;
             let topic_suffix = match input.topic.name.split_once('/') {
                 Some((_, topic_suffix)) => topic_suffix,
                 None => return Ok(vec![]),
             };
-
             let out_topic = Topic::new(&format!("aws/td/{topic_suffix}"))?;
 
             let output = MqttMessage::new(&out_topic, payload);
             self.size_threshold.validate(&output)?;
             Ok(vec![(output)])
+        } else {
+            Ok(vec![])
+        }
+    }
+
+    fn try_convert_te_topics(
+        &mut self,
+        source: EntityTopicId,
+        channel: Channel,
+        input: &MqttMessage,
+    ) -> Result<Vec<MqttMessage>, ConversionError> {
+        match &channel {
+            Channel::Measurement { measurement_type } => {
+                self.convert_message(input, source, measurement_type)
+            }
+
+            Channel::Event { event_type } => self.convert_message(input, source, event_type),
+
+            Channel::Alarm { alarm_type } => self.convert_message(input, source, alarm_type),
+            _ => Ok(vec![]),
+        }
+    }
+
+    fn convert_message(
+        &mut self,
+        input: &MqttMessage,
+        source: EntityTopicId,
+        telemetry_type: &String,
+    ) -> Result<Vec<MqttMessage>, ConversionError> {
+        let payload = self.with_time_stamp(input)?;
+        let source = normalize_name(source);
+        let out_topic = match input.topic.name.split('/').collect::<Vec<_>>()[..] {
+            [_, "device", _, _, _, "m", _] => {
+                Topic::new_unchecked(&format!("aws/td/{source}/m/{telemetry_type}"))
+            }
+            [_, "device", _, _, _, "e", _] => {
+                Topic::new_unchecked(&format!("aws/td/{source}/e/{telemetry_type}"))
+            }
+            [_, "device", _, _, _, "a", _] => {
+                Topic::new_unchecked(&format!("aws/td/{source}/a/{telemetry_type}"))
+            }
```

Comment:
The `device` prefix must not be enforced.

```suggestion
            [_, _, _, _, _, "m", _] => {
                Topic::new_unchecked(&format!("aws/td/{source}/m/{telemetry_type}"))
            }
            [_, _, _, _, _, "e", _] => {
                Topic::new_unchecked(&format!("aws/td/{source}/e/{telemetry_type}"))
            }
            [_, _, _, _, _, "a", _] => {
                Topic::new_unchecked(&format!("aws/td/{source}/a/{telemetry_type}"))
            }
```

Reply:
fixed